sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -640,8 +664,8 @@ def read(self, size: Optional[int] = -1) -> T:
 
         data = stream.getvalue()
         if self._encoding:
-            return data.decode(self._encoding)
-        return data
+            return data.decode(self._encoding)  # type: ignore [return-value]
```

Comment:
I am going to make this comment on this but it applies to all the returns for any function that `-> T`
This is why I have this ignore: https://github.com/python/mypy/issues/11536

MyPy does _NOT_ know how to analyze this as expected...